manifest/src/lib.rs

```diff
@@ -0,0 +1,106 @@
+use anyhow::Result;
+use npm_package_json::Package;
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+use std::{ffi::OsStr, fs};
+use strum::{Display, EnumIter};
+
+#[macro_use]
+extern crate lazy_static;
+
+lazy_static! {
+    static ref HASHMAP: HashMap<&'static str, ManifestType> = {
+        let mut m = HashMap::new();
+        m.insert("Cargo.toml", ManifestType::Cargo);
+        m.insert("package.json", ManifestType::Npm);
+        m
+    };
+}
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct Manifest {
+    pub manifest_type: ManifestType,
+    pub number_of_dependencies: usize,
+    pub name: String,
+    pub description: Option<String>,
+}
+
+#[derive(Display, Clone, Copy, PartialEq, Eq, Debug, EnumIter)]
+pub enum ManifestType {
+    Npm,
+    Cargo,
+}
+
+pub fn get_manifests<P: AsRef<Path>>(path: P) -> Result<Vec<Manifest>> {
+    let manifest_paths = get_manifest_paths(path)?;
+    let manifests = manifest_paths
+        .iter()
+        .map(|manifest_path| {
+            let filename = manifest_path
+                .file_name()
+                .and_then(|name| name.to_str())
+                .unwrap_or_default();
+
+            let manifest_type = HASHMAP[filename];
+
+            match manifest_type {
+                ManifestType::Cargo => parse_cargo_manifest(manifest_path),
+                ManifestType::Npm => parse_npm_manifest(manifest_path),
+            }
+        })
+        .filter_map(std::result::Result::ok)
+        .collect::<Vec<_>>();
+    Ok(manifests)
+}
+
+fn get_manifest_paths<P: AsRef<Path>>(path: P) -> Result<Vec<PathBuf>> {
```

Comment:
It seems like this function can be combined with `get_manifests()` to do all work in one swoop. Generally, `contains_key` checks can be avoided in favour of `map.get()` which yields an `Option<V>`.

Reply:
You're right, done

Reply:
A separate `get_manifest_paths` function might be helpful to others that want to use this manifest crate as a dependency.